app/utilities/featureToggleParser/featureToggleParser.factory.spec.js

```diff
@@ -2,8 +2,303 @@
+  
+      });
+  
+      it('should return same object when the parseConfigObject is called with object without enableDates property', function() {
+  
+          newConfig = {
+              "configSettings": "abc"
+          };
+  
+          result = featureToggleParser.parseConfigObject(newConfig);
+  
+          expect(result.configSettings).toBe("abc");
+  
+      });
+  
+  });
+  
+  describe('parseToggleObject()', function() {
+  
+      var result, toggles, input, output;
+      var emptyObject = {};
+      var InvalidDate = "Invalid date";
+  
+      beforeEach(function() {
+  
+          input = {
+              "flag1": false,
+              "flag2": true,
+              "flag3": "19911001",
+              "flag4": "true",
+              "flag5": 12312,
+              "flag6": "2017-03-10",
+              "flag7": "abc",
+              "flag8": "1234567890",
+              "flag9": "FALSE",
+              "list": {
+                  "fruit": [
+                      "apples",
+                      "bananas",
+                      "grapefruit"
+                  ]
+              },
+              "validObj": {
+                  "isValid": true
+              }
+          };
+  
+          output = {
+              "flag1": false,
+              "flag2": true,
+              "flag3": 'Tue Oct 01 1991 00:00:00 GMT+0530',
+              "flag4": true,
+              "flag6": 'Fri Mar 10 2017 00:00:00 GMT+0530',
+              "flag9": false
+          };
+  
+      });
+  
+      it('should return empty object on passing an empty JSON object', function() {
+  
+          toggles = {};
+  
+          result = featureToggleParser.parseToggleObject(toggles);
+  
+          expect(result).toEqual(emptyObject);
+  
+      });
+  
+      it('should return empty object on passing an empty string', function() {
+  
+          toggles = "";
+  
+          result = featureToggleParser.parseToggleObject(toggles);
+  
+          expect(result).toEqual(emptyObject);
+  
+      });
+  
+      it('should return empty object on passing an array', function() {
+  
+          toggles = [1, 2, 3, 4, 5];
+  
+          result = featureToggleParser.parseToggleObject(toggles);
+  
+          expect(result).toEqual(emptyObject);
+  
+      });
+  
+      it('should return empty object on passing a number', function() {
+  
+          toggles = 5;
+  
+          result = featureToggleParser.parseToggleObject(toggles);
+  
+          expect(result).toEqual(emptyObject);
+  
+      });
+  
+      it('should return empty object on passing a boolean value', function() {
+  
+          toggles = true;
+  
+          result = featureToggleParser.parseToggleObject(toggles);
+  
+          expect(result).toEqual(emptyObject);
+  
+      });
+  
+      it("should test the returned property be 'moment' type when passing property as date string in `YYYYMMDD` format", function() {
+  
+          result = featureToggleParser.parseToggleObject(input);
```

Comment:
the code does not test if it is a moment object when passing as a string, it attempts to convert to a moment object.  Your test case tests if the resulting value is a moment object -- thus the test description is misleading.